### PR TITLE
service: update error types to be more logicless, simplify validations

### PR DESCRIPTION
**Pull request by ilgooz**
depends on #383, it should be merged before this one. this PR will be rebased from dev after depended PR makes its way through dev.

please review this PR by commits.

### CODE REVIEW

File: api/execute_executor.go [hunk 1]
```diff
@@ -47,9 +47,24 @@ func (e *taskExecutor) checkServiceStatus(s *service.Service) error {
...
+	}
+	warnings := s.ValidateParametersSchema(serviceTask.Inputs, taskInputs)
+	if len(warnings) > 0 {
+		return "", &service.InvalidTaskInputError{
+			TaskKey:  taskKey,
+			Warnings: warnings,
+		}
+	}
+
+	exc, err := execution.Create(s, taskKey, taskInputs, tags)
```
Comment (antho1404) [inline api/execute_executor.go]: I'm curious to have some arguments to move the validation outside of the create. For me the validation here is really related to the execution so it make sense that the execution package does the validation. Same point for the event and the results

Comment (ilgooz) [reply api/execute_executor.go]: I can't 100% agree with this. calling methods of specialized packages like _execution_ with invalid inputs seemed to me a bit messy. I believe that these validations should be done in the api package, a place that where we compose lots of types and different internal packages. This compositions requires some validations and I think doing this in package api feels natural. For _Create()_, to me, there is no difference between checking if the given service is nil -which doesn't make sense- and checking the existence of task. 

But I'm not strongly against for your version, @NicolasMahe what do you think about this?

Comment (NicolasMahe) [reply api/execute_executor.go]: Also, I thought you will put the `validate` functions in the service package and they returns directly the error, like in https://github.com/mesg-foundation/core/issues/348#issuecomment-416183529

Comment (ilgooz) [reply api/execute_executor.go]: I'll do it in another PR

**Pull request merged**